src/metabase/models/collection.clj

```diff
@@ -947,18 +947,24 @@
 
   Practically, use `user-or-site` = `:site` when insert or update the name in database,
   and `:user` when we need the name for displaying purposes"
-  [first-name last-name user-or-site]
+  [first-name last-name email user-or-site]
   {:pre [(#{:user :site} user-or-site)]}
   (if (= :user user-or-site)
-    (tru "{0} {1}''s Personal Collection" first-name last-name)
-    (trs "{0} {1}''s Personal Collection" first-name last-name)))
+    (cond
+      (and first-name last-name) (tru "{0} {1}''s Personal Collection" first-name last-name)
+      :else                      (tru "{0}''s Personal Collection" (or first-name last-name email)))
+    (cond
+      (and first-name last-name) (trs "{0} {1}''s Personal Collection" first-name last-name)
+      :else                      (trs "{0}''s Personal Collection" (or first-name last-name email)))))
```

Comment:
A user's personal collection name is derived from their first and last names. If either or both are 'nil', the old implementation would return `null null's Personal Collection`. Now it'll fall back to first OR last name OR email.